src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java

```diff
@@ -143,8 +143,21 @@ public void load() throws IOException {
      */
     public void persist() throws IOException {
         final Path path = Paths.get(fileName);
-        final Path directory = path.getParent();
+        Path directory = path.getParent();
+
         if (directory != null) {
+            if (Files.isSymbolicLink(directory)) {
+                final Path actualDir = Files.readSymbolicLink(directory);
```

Comment:
What if a symbolic points to another symbolic? Will this code follow until it gets to a physical file/directory or a non-existent location?

Reply:
> What if a symbolic points to another symbolic? Will this code follow until it gets to a physical file/directory or a non-existent location?

The modified code is doing it to the best of investigation.

I am using the below script:
```
#!/bin/bash

# Create a directory 't' which will be the final target directory
mkdir -p t

# Create a symbolic link 'link1' pointing to 't'
ln -snf ./t ./link1

# Create another symbolic link 'target' pointing to 'link1'
ln -snf ./link1 ./target

# Path to the Checktyle JAR file
CHECKSTYLE_JAR="/Users/piyushkumarsadangi/Development/checkstyle/checkstyle/target/checkstyle-10.12.7-SNAPSHOT-all.jar"

# Path to the Java source file to be checked, relative to the script's location.
JAVA_SOURCE="src/Main.java"

# Path to the Checkstyle configuration file, relative to the script's location.
CHECKSTYLE_CONFIG="checkstyle.xml"

# Running Checkstyle
java -cp "$CHECKSTYLE_JAR" com.puppycrawl.tools.checkstyle.Main "$JAVA_SOURCE" -c "$CHECKSTYLE_CONFIG"
```

Reply:
> What if a symbolic points to another symbolic? Will this code follow until it gets to a physical file/directory or a non-existent location?

is it necessary to do that? i mean the modified code does it, but leading to pitest failure. but, the code for which the review comment was written, doesn't do that.

Reply:
> > What if a symbolic points to another symbolic? Will this code follow until it gets to a physical file/directory or a non-existent location?
> 
> is it necessary use-case to be incorporated i.e. one symlink pointing to another symlink? i mean the modified code does it, but leading to pitest failure. but, the code for which the review comment was written, doesn't do that.

Reply:
May be we should wait for next 10 years , before defect on this will be reported to us ?
Crazy edge case.

Reply:
@rnveach, do you insist to fix it now?

Reply:
The latest push does it to my understanding. `final Path actualDir = directory.toRealPath();` helps to resolve multiple symlink.

Reply:
> The latest push does it 

Which test case shows it is fixed?